test/e2e/storage_performant_policy.go

```diff
@@ -0,0 +1,289 @@
+)
+
+var (
+	waitPollInterval = time.Second
+)
+
+// newClientConfigForTest returns a config configured to connect to the api server
+func newClientConfigForTest() (*rest.Config, error) {
+	loader := clientcmd.NewDefaultClientConfigLoadingRules()
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loader, &clientcmd.ConfigOverrides{ClusterInfo: api.Cluster{InsecureSkipTLSVerify: true}})
+	config, err := clientConfig.ClientConfig()
+	if err == nil {
+		g.GinkgoLogr.Info("Found configuration for", "host", config.Host)
+	}
+	return config, err
+}
+
+var _ = g.Describe("[sig-storage][OCPFeatureGate:StoragePerformantPolicy] Storage Performant Policy", g.Label("Conformance"), g.Label("Parallel"), func() {
+	var kubeClient *kubernetes.Clientset
+
+	g.BeforeEach(func() {
+		kubeConfig, err := newClientConfigForTest()
+		if err != nil {
+			g.Fail(fmt.Sprintf("Failed to get kubeconfig: %v", err))
+		}
+		kubeClient = kubernetes.NewForConfigOrDie(rest.AddUserAgent(kubeConfig, clientName))
+	})
+
+	g.Context("namespace has fsgroup change policy label", func() {
+		var nsObj *v1.Namespace
+		g.BeforeEach(func(ctx context.Context) {
+			nsObj = &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "fsgroup-policy-test-",
+					Labels: map[string]string{
+						fsGroupPolicyLabel: string(v1.FSGroupChangeOnRootMismatch),
+					},
+				},
+			}
+			testContext, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+
+			var err error
+			nsObj, err = kubeClient.CoreV1().Namespaces().Create(testContext, nsObj, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				g.Fail(fmt.Sprintf("Failed to create test namespace: %v", err))
+			}
+			g.GinkgoLogr.Info("Created namespace with label storage.openshift.io/fsgroup-change-policy=OnRootMismatch", "namespace", nsObj.Name)
+		})
+
+		g.AfterEach(func(ctx context.Context) {
+			testContext, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+			cleanupNamespace(testContext, kubeClient, nsObj.Name)
+		})
+
+		g.It("it should set fsgroup change policy to OnRootMismatch if pod has none", func(ctx context.Context) {
+			testContext, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+			pod := getPod(nsObj.Name)
+
+			createdPod, err := kubeClient.CoreV1().Pods(nsObj.Name).Create(testContext, pod, metav1.CreateOptions{})
+			if err != nil {
+				g.Fail(fmt.Sprintf("Failed to create test pod: %v", err))
+			}
+			g.GinkgoLogr.Info("Created pod in namespace", "pod", createdPod.Name, "namespace", nsObj.Name)
+
+			// Wait for pod to be scheduled
+			err = wait.PollUntilContextCancel(testContext, waitPollInterval, false, func(ctx context.Context) (bool, error) {
+				p, err := kubeClient.CoreV1().Pods(nsObj.Name).Get(ctx, createdPod.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				return p.Status.Phase != v1.PodPending, nil
```

Comment:
Some log with `waiting for pod %s to be Running, got %s` would be helpful. If the test needs to wait for Running at all.

Reply:
nah, the test doesn't need to wait for pods to be running. I removed that part. It was left as an oversight in other test.